wpilib/src/lib.rs

```diff
@@ -4,11 +4,26 @@
 // <https://www.gnu.org/licenses/> for a copy.
 
 #![feature(concat_idents)]
-
-#[macro_use]
+#![macro_use]
 extern crate lazy_static;
+extern crate wpilib_sys;
+
+mod robot_base;
+mod pdp;
+mod sensor_util;
+
+pub mod dio;
+pub mod ds;
+pub mod time;
+pub mod joystick;
+pub mod pneumatics;
+pub mod serial;
+pub mod spi;
+
+pub use self::dio::{DigitalInput, DigitalOutput};
+pub use self::robot_base::*;
+pub use self::pdp::*;
 
-mod hal;
-pub use hal::{report_usage, report_usage_extras, HalError, HalMaybe, HalResult};
-mod wpilib;
-pub use wpilib::*;
+pub use wpilib_sys::bindings::*;
+pub use wpilib_sys::hal_call::{HalMaybe, HalError, HalResult};
+pub use wpilib_sys::usage::{report_usage, report_usage_extras};
```

Comment:
I'm not sure you actually want to reexport these here. You _definitely_ don't want to reexport the FFI bindings.

Reply:
I'm not sure about the usage. It seems like something someone might actually use on their robot, but they have more use in wpilib-sys.

Reply:
If they really want to use it, having them usable only from the `wpilib_sys` crate would make it clear that it's not the intended API. Also we probably don't want to clutter the docs with all the HAL functions.